### PR TITLE
Add onCancel handler

### PR DESCRIPTION
**Pull request by djmccormick**

### CODE REVIEW

File: src/action.js [hunk 1]
```diff
@@ -246,6 +246,25 @@ export default class Action extends Emitter {
...
+   * @param {Function} callback
+   * @param {any} scope
+   * @return {Action} self
+   */
+  onCancel (callback, scope) {
+    if (this.is(States.cancelled)) {
+      callback(this.payload, scope)
+    } else {
+      this.on('cancel', callback.bind(scope))
+    }
```
Comment (nhunzaker) [inline src/action.js]: Awesome! The emitter supports `once`, is there a case where this would ever be emitted more than once?

**Comment by nhunzaker** [comment]
@ltk Any idea why coveralls is failing here?

**Comment by djmccormick** [comment]
Locally I didn't get errors with `npm test`:

    djmccormick at forge in ~/Projects/microcosm on feature-oncancel
    ± npm test
    
    > microcosm@10.0.0-beta6 test /Users/djmccormick/Projects/microcosm
    > ava
    
    
       127 passed

Looks like a coverage problem in CircleCI. I can take a stab at testing.

File: src/action.js (hunk 1 above)
Comment (djmccormick) [inline src/action.js]: Done, thanks!

**Comment by ltk** [comment]
Nice, :+1:. Ping me if you're still having issues with Circle/Coveralls and I can take a look.

**Comment by nhunzaker** [comment]
@djmccormick All good. I'll merge this.

**Comment by nhunzaker** [comment]
@djmccormick I'll publish a new beta after we sort out >>>>>>> Stashed changes

**Comment by nhunzaker** [comment]
@ltk I think it was my fault. Let's see what happens next time.

**Pull request merged**